transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringBufferRing.java

```diff
@@ -0,0 +1,452 @@
+            recycleBuffer(bid);
+        }
+
+        @Override
+        protected byte _getByte(int index) {
+            return userspaceBuffer.getByte(index);
+        }
+
+        @Override
+        protected short _getShort(int index) {
+            return userspaceBuffer.getShort(index);
+        }
+
+        @Override
+        protected short _getShortLE(int index) {
+            return userspaceBuffer.getShortLE(index);
+        }
+
+        @Override
+        protected int _getUnsignedMedium(int index) {
+            return userspaceBuffer.getUnsignedMedium(index);
+        }
+
+        @Override
+        protected int _getUnsignedMediumLE(int index) {
+            return userspaceBuffer.getUnsignedMediumLE(index);
+        }
+
+        @Override
+        protected int _getInt(int index) {
+            return userspaceBuffer.getInt(index);
+        }
+
+        @Override
+        protected int _getIntLE(int index) {
+            return userspaceBuffer.getIntLE(index);
+        }
+
+        @Override
+        protected long _getLong(int index) {
+            return userspaceBuffer.getLong(index);
+        }
+
+        @Override
+        protected long _getLongLE(int index) {
+            return userspaceBuffer.getLongLE(index);
+        }
+
+        @Override
+        protected void _setByte(int index, int value) {
+            userspaceBuffer.setByte(index, value);
+        }
+
+        @Override
+        protected void _setShort(int index, int value) {
+            userspaceBuffer.setShort(index, value);
+        }
+
+        @Override
+        protected void _setShortLE(int index, int value) {
+            userspaceBuffer.setShortLE(index, value);
+        }
+
+        @Override
+        protected void _setMedium(int index, int value) {
+            userspaceBuffer.setMedium(index, value);
+        }
+
+        @Override
+        protected void _setMediumLE(int index, int value) {
+            userspaceBuffer.setMediumLE(index, value);
+        }
+
+        @Override
+        protected void _setInt(int index, int value) {
+            userspaceBuffer.setInt(index, value);
+        }
+
+        @Override
+        protected void _setIntLE(int index, int value) {
+            userspaceBuffer.setIntLE(index, value);
+        }
+
+        @Override
+        protected void _setLong(int index, long value) {
+            userspaceBuffer.setLong(index, value);
+        }
+
+        @Override
+        protected void _setLongLE(int index, long value) {
+            userspaceBuffer.setLongLE(index, value);
+        }
+
+        @Override
+        public int capacity() {
+            return maxCapacity();
+        }
+
+        @Override
+        public ByteBuf capacity(int newCapacity) {
+            if (newCapacity <= maxCapacity()) {
+                this.maxCapacity(newCapacity);
+                return this;
+            }
+            ByteBuf newByteBuffer = byteBufAllocator.directBuffer(newCapacity);
```

Comment:
if newCapacity > maxCapacity throw an `IllegalArgumentException`. This is the API contract that is documented in the ByteBuf javadocs